api/src/middleware/__tests__/assessmentsRouter.ts

```diff
@@ -1,18 +1,242 @@
+        {
+          curriculum_assessment: curriculumAssessment,
+          program_assessment: programAssessment,
+          submissions_summary: assessmentSubmissionsSummary,
+        },
+      ];
+      // mock response from (function that gets a list of programs the user is enrolled in) to include one program
+      mockPrincipalEnrolledPrograms.mockResolvedValue([4]);
+      // mock response from (call a function that returns the permission of the user for each program (participant/facilitator)) to respond with participant for that one program
+      mockGetProgramIdByProgramAssessmentId.mockResolvedValue([
+        exampleAssessmentId,
+      ]);
+      mockFindRoleInProgram.mockResolvedValue({ title: 'participant' });
+      // mock response from (get a list of program assessments for each program the user is enrolled in) to respond with a list of program assessments
+      mockGetAssessmentsForProgram.mockResolvedValue(programAssessment);
+      // mock responses from (get a list of curriculum assessments that correspond to each program assessment) to respond with the corresponding curriculum assessment for each program assessment in that previous list
+      mockGetCurriculumAssessmentById.mockResolvedValue(curriculumAssessment);
+      // mock responses from (call a (mock) function that gets the participant assessment summary for each program assessment where the user is a participant of that program) to respond with the assessment summary for every program assessment that participant has submitted for previously
+      mockGetAssessmentSubmissionsSummary.mockResolvedValue(
+        assessmentSubmissionsSummary
+      );
+      appAgent
+        .get('/')
+        .expect(
+          200,
+          collectionEnvelope(
+            participantAssessmentListResponse,
+            participantAssessmentListResponse.length
+          ),
+          err => {
+            // call a (mock) function that gets a list of programs the user is enrolled in
+            expect(mockPrincipalEnrolledPrograms).toHaveBeenCalledWith(
+              enrolledParticipantPrincipalId
+            );
+            // call a function that returns the permission of the user for each program (participant/facilitator)
+            expect(mockGetProgramIdByProgramAssessmentId).toHaveBeenCalledWith(
+              programAssessmentId
+            );
+            expect(mockFindRoleInProgram).toHaveBeenCalledWith(
+              programId,
+              enrolledParticipantPrincipalId
+            );
+            //get a list of program assessments for each program the user is enrolled in) to respond with a list of program assessments
+            expect(mockGetAssessmentsForProgram).toHaveBeenCalledWith(
+              programId
+            );
+            // get a list of curriculum assessments that correspond to each program assessment
+            expect(mockGetCurriculumAssessmentById).toHaveBeenCalledWith(
+              exampleAssessmentId,
```

Comment:
This first parameter should be `programAssessment.curriculum_id` instead. Even if it's the same value, it makes more sense here, since we should be getting back an array that includes `programAssessment` from the `getAssessmentsForProgram` function.